### PR TITLE
virsh_cpu_stats: add new cases and more strict result checking

### PR DESCRIPTION
**Pull request by czhang0**
changes in this commit:
1. add powpc cases
2. add case with no vm start
3. several negative cases
4. add error message checking for negative cases

Signed-off-by: cuzhang <cuzhang@redhat.com>

### CODE REVIEW

**Comment by czhang0** [comment]
Locally test run result is given following:
avocado run --vt-type libvirt local..cpu_stats
JOB ID     : 9b2be00849ec580cf0239c16a8ae1ef58e4d1cdd
JOB LOG    : /root/avocado/job-results/job-2018-02-28T10.19-9b2be00/job.log
 (01/15) type_specific.local.virsh.cpu_stats.positive_test.option1: PASS (16.72 s)
 (02/15) type_specific.local.virsh.cpu_stats.positive_test.option2: PASS (21.67 s)
 (03/15) type_specific.local.virsh.cpu_stats.positive_test.option3: PASS (21.69 s)
 (04/15) type_specific.local.virsh.cpu_stats.positive_test.option4: PASS (21.72 s)
 (05/15) type_specific.local.virsh.cpu_stats.positive_test.option5: PASS (21.71 s)
 (06/15) type_specific.local.virsh.cpu_stats.positive_test.option6: PASS (21.97 s)
 (07/15) type_specific.local.virsh.cpu_stats.negative_test.vm_is_not_start: PASS (1.51 s)
 (08/15) type_specific.local.virsh.cpu_stats.negative_test.no_domain: PASS (21.66 s)
 (09/15) type_specific.local.virsh.cpu_stats.negative_test.unexpect_domain: PASS (21.69 s)
 (10/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_domain: PASS (21.69 s)
 (11/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_start_option: PASS (21.66 s)
 (12/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_start_negative: PASS (21.74 s)
 (13/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_start_alpha: PASS (21.73 s)
 (14/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_start_whitespace: PASS (21.70 s)
 (15/15) type_specific.local.virsh.cpu_stats.negative_test.invalid_start_none: PASS (21.68 s)
RESULTS    : PASS 15 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 305.03 s

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 1]
```diff
@@ -46,3 +59,18 @@
...
+                - invalid_start_alpha:
+                    cpu_stats_options = "--start abc"
+                    error_msg = "Numeric value 'abc' for <start> option is malformed or out of range"
+                - invalid_start_whitespace:
+                    cpu_stats_options = "--start ' '"
+                    error_msg = "Numeric value ' ' for <start> option is malformed or out of range"
+                - invalid_start_none:
+                    cpu_stats_options = "--start ''"
+                    error_msg = "Numeric value '' for <start> option is malformed or out of range"
+>>>>>>> virsh_cpu_stats: add new cases and more strict result checking
```
Comment (hahakiki2010) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Pls fixup the conflict first

File: libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py [hunk 2]
```diff
@@ -82,6 +87,7 @@ def run(test, params, env):
                 vm = env.get_vm(vm_ref)
                 cgpath = utils_cgroup.resolve_task_cgroup_path(
                     vm.get_pid(), "cpuacct")
+                logging.debug("cgpath: %s" % cgpath)
```
Comment (hahakiki2010) [inline libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py]: ```logging.debug("cgpath: %s", cgpath) ``` is better

File: libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py [hunk 3]
```diff
@@ -73,6 +74,10 @@ def run(test, params, env):
     if status_error == "yes":
         if status == 0:
             test.fail("Run successfully with wrong command!")
+        else:
+            # Check error message is expected
+            if not re.search(error_msg, cmd_result.stderr.strip()):
+                test.TestFail("Error message is not expected!")
```
Comment (hahakiki2010) [inline libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py]: shall be test.fail

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 4]
```diff
@@ -3,17 +3,21 @@
     take_regular_screendumps = "no"
     cpu_stats_vm_ref = "name"
     cpu_stats_options = ""
+    start_vm = "yes"
     variants:
         - positive_test:
             status_error = "no"
             variants:
                 - option1:
+                    no ppc64,ppc64le
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Why to exclude ppc? These options work on ppc.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 5]
```diff
@@ -3,17 +3,21 @@
...
                     cpu_stats_options = "--total --start 0 --count 3"
                 - option2:
                     cpu_stats_options = "--total"
                 - option3:
+                    no ppc64,ppc64le
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Same as above

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 6]
```diff
@@ -3,17 +3,21 @@
...
                     cpu_stats_options = "--start 1 --count 100"
                 - option4:
+                    no ppc64,ppc64le
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Same as above

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 7]
```diff
@@ -27,15 +31,24 @@
                     cpu_stats_options = "--total --count 10"
                 - paused_option:
                     paused_after_start_vm = "yes"
+                - test_ppc:
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: This case can also work on x86, no need to make it ppc specific.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg [hunk 8]
```diff
@@ -46,3 +59,18 @@
                     cpu_stats_options = "--start -1 --count -5"
                 - invalid_start_option5:
                     cpu_stats_options = "--start 1.5"
+                    cpu_stats_options = "--start 65555234"
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Remove one 'cpu_stats_options', please

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg (hunk 1 above)
Comment (czhang0) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: Fixed.

File: libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py (hunk 2 above)
Comment (czhang0) [reply libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py]: Ok

File: libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py (hunk 3 above)
Comment (czhang0) [reply libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py]: Yes, code need to be align to recent change of 'remove autotest dependency'.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg (hunk 7 above)
Comment (czhang0) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: I misunderstand it. I'll modify it.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg (hunk 4 above)
Comment (czhang0) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: I misunderstand it. I'll modify it.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg (hunk 5 above)
Comment (czhang0) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: I misunderstand it. I'll modify it.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg (hunk 6 above)
Comment (czhang0) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_stats.cfg]: I misunderstand it. I'll modify it.

**Pull request merged**